test/odbc/psqlodbc/datetime.cpp

```diff
@@ -0,0 +1,369 @@
+  };
+  const int NUM_OF_DATA = INSERTED_DATA.size();
+
+  // insertString initialization
+  string insertString{};
+  string comma{};
+  for (int i = 0; i < NUM_OF_DATA; i++) {
+    insertString += comma + "(\'" + INSERTED_PK[i] + "\',\'" + INSERTED_DATA[i] + "\')";
+    comma = ",";
+  }
+
+  const vector<string> OPERATIONS_QUERY = {
+    COL1_NAME + "=" + COL2_NAME,
+    COL1_NAME + "<>" + COL2_NAME,
+    COL1_NAME + "<" + COL2_NAME,
+    COL1_NAME + "<=" + COL2_NAME,
+    COL1_NAME + ">" + COL2_NAME,
+    COL1_NAME + ">=" + COL2_NAME
+  };
+
+  // initialization of expected_results
+  vector<vector<char>> expected_results = {};
+
+  for (int i = 0; i < NUM_OF_DATA; i++) {
+    expected_results.push_back({});
+    const char *date_1 = INSERTED_PK[i].data();
+    const char *date_2 = INSERTED_DATA[i].data();
+    expected_results[i].push_back(strcmp(date_1, date_2) == 0 ? '1' : '0');
+    expected_results[i].push_back(strcmp(date_1, date_2) != 0 ? '1' : '0');
+    expected_results[i].push_back(strcmp(date_1, date_2) < 0 ? '1' : '0');
+    expected_results[i].push_back(strcmp(date_1, date_2) <= 0 ? '1' : '0');
+    expected_results[i].push_back(strcmp(date_1, date_2) > 0 ? '1' : '0');
+    expected_results[i].push_back(strcmp(date_1, date_2) >= 0 ? '1' : '0');
+  }
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS);
+  insertValuesInTable(ServerType::PSQL, TABLE_NAME, insertString, NUM_OF_DATA);
+  testComparisonOperators(ServerType::PSQL, TABLE_NAME, COL1_NAME, COL2_NAME, INSERTED_PK, INSERTED_DATA, 
+    OPERATIONS_QUERY, expected_results);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+// inserted values differ that of expected?
+TEST_F(PSQL_Datatypes_Datetime, Comparison_Functions) {
+  const vector<string> INSERTED_DATA = {
+    "1753-01-01 00:00:000",
+    "2011-04-15 16:44:09.000",
+    "9999-12-31 23:59:59.997"
+  };
+
+  const vector<string> EXPECTED_RESULTS = {
+    "1753-01-01 00:00:00",
+    "2011-04-15 16:44:09",
+    "9999-12-31 23:59:59.997"
+  };
+  const int NUM_OF_DATA = INSERTED_DATA.size();
+  
+  // insertString initialization
+  string insertString{};
+  string comma{};
+  for (int i = 0; i < NUM_OF_DATA; i++) {
+    insertString += comma + "(" + std::to_string(i) + ",\'" + INSERTED_DATA[i] + "\')";
+    comma = ",";
+  }
+
+  const vector<string> OPERATIONS_QUERY = {
+    "MIN(" + COL2_NAME + ")",
+    "MAX(" + COL2_NAME + ")"
+  };
+  const int NUM_OF_OPERATIONS = OPERATIONS_QUERY.size();
+
+  // initialization of expected_results
+  vector<string> expected_results = {};
+  int min_expected = 0, max_expected = 0;
+
+  for (int i = 1; i < NUM_OF_DATA; i++) {
+    const char *currMin = EXPECTED_RESULTS[min_expected].data();
+    const char *currMax = EXPECTED_RESULTS[max_expected].data();
+    const char *curr = EXPECTED_RESULTS[i].data();
+
+    min_expected = strcmp(curr, currMin) < 0 ? i : min_expected;
+    max_expected = strcmp(curr, currMax) > 0 ? i : min_expected;
```

Comment:
```suggestion
    max_expected = strcmp(curr, currMax) > 0 ? i : max_expected;
```
Had a typo on my end when creating this.

Reply:
Fixed.